### PR TITLE
feat(question): select category

### LINKED ISSUE #2963: [QuestionCreate] Select category
It should be possible to define the Catgories which are appropriate for the content based on a predefined set of tags. Look at Research/How-to for an example.

Stretch. Refactor our current implementation to remove duplicate collections.

### PR DESCRIPTION
**Pull request by oktomus**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Add categories to questions, like Research and How-To.

⚠️ I think work is required on the firebase side to add the categories, because I don't think there is an administration for the website?

## Git Issues

Closes #2963

## Screenshots/Videos

rename tag

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: src/stores/QuestionCategories/questionCategories.store.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { observable, action, makeObservable, computed } from 'mobx'
+import { arrayToJson } from 'src/utils/helpers'
+import { ModuleStore } from '../common/module.store'
+import type { RootStore } from '..'
+import type { IQuestionCategory } from '../../models/questionCategories.model'
+
+export class QuestionCategoriesStore extends ModuleStore {
```
Comment (oktomus) [inline src/stores/QuestionCategories/questionCategories.store.tsx:7]: This is a pure copy of ResearchCategoriesStore, but with renaming.

**Review by evakill — COMMENTED** [review]
hey @oktomus, thanks for you contribution!! 

code LGTM and follows existing patterns. as you point out, we don't have an FE interface for writing categories. i think this is/was part of the road map, which is why we decided to make tags and categories backed by DB collections. i will check in with the moderators about if/when we will build out this feature fully!

looks like a rebase is needed, which is our bad for the late review 🙈 
after that will approve

**Comment by oktomus** [comment]
Thanks for the review @evakill! I did the changes you suggested, rebased and fixed the conflicts, tested the widget, and ran `yarn format:style`. 

Only thing missing is adding categories in the DB.

**Comment by evakill** [comment]
aha - one more failing test. you need to mock the `questionCategoriesStore` in this test - `src/pages/Question/question.routes.test.tsx`

see `src/pages/Research/research.routes.test.tsx` as an example! 

i am adding some test data to the dev DB so we can test locally and on the preview branch

**Comment by oktomus** [comment]
Ok should be good now :) Thanks for the heads up, I'm still learning about all these web techs ahah

**Comment by oktomus** [comment]
I think the failed test above has nothing to do with the pull request, it's a time out, should we run it again?

**Comment by evakill** [comment]
@oktomus yes the firefox tests are so flakey 😭 

merging! thanks again

**Review by evakill — APPROVED** [review]

**Pull request merged**